### PR TITLE
Move logic of displaying number of friends to decorator

### PR DESCRIPTION
**Pull request by fugakkbn**
follow up: #74 

It would be better to write the display logic in the Decorator than in the View.

### CODE REVIEW

**Comment by unasuke** [comment]
> It would be better to write the display logic in the Decorator than in the View.

@fugakkbn Reasonable. But the logic should be implemented in UserDecorator, shouldn't it?

**Comment by fugakkbn** [comment]
@unasuke 
You're correct.
For now, `#number_of_friends` is used only in ProfileView. So I thought ProfileDecorator would be a good choice.

But I agree that it is the responsibility of the UserDecorator.
If we prefer to implement it in UserDecorator, I will do so. How do you think?

**Comment by unasuke** [comment]
UserDecorator, please!

**Comment by fugakkbn** [comment]
Sorry, I closed this PR by mistake, could you reopen?

**Comment by fugakkbn** [comment]
@unasuke UserDecorator e548902

File: app/decorators/user_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module UserDecorator
+  def number_of_friends(friends)
```
Comment (unasuke) [inline app/decorators/user_decorator.rb:4]:
```suggestion
  def number_of_friends
```

`User` model knows its friends already.

**Review by unasuke — COMMENTED** [review]
I noticed `current_user` doesn't preload its friends record now...ah...

**Comment by unasuke** [comment]
* A: preload current user's friends in SessionHepler
* B: Create `@user` instance variable in ProfilesController

🤔

**Comment by hayaokimura** [comment]
I think another choice is exists.
Create `@number_of_friends` instande variable in ProfilesController  like this
```ruby
@number_of_friends = current_user.preload(:friends).number_of_friends
```

**Comment by fugakkbn** [comment]
We want to preload `friends` for the convenience of Profile#index.
So it would be better to prepare @user in Profile#index.

**Review by unasuke — APPROVED** [review]
👍 👍 👍 👍

**Pull request merged**